CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
 # Contributing Guidelines
 
-Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
-documentation, we greatly value feedback and contributions from our community.
+Thank you for your interest in contributing to the Android Amplify
+framework. Whether it's a bug report, new feature, correction, or
+additional documentation, the project maintainers at AWS greatly value
+your feedback and contributions.
 
-Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
-information to effectively respond to your bug report or contribution.
+Please read through this document before submitting any issues or pull
+requests. Doing so will help to ensure that the project maintainers have
+all information neessary to effectively respond to your bug report or
+contribution.
 
+- [Contributing Guidelines](#contributing-guidelines)
+  * [Amplify History and Ethos](#amplify-history-and-ethos)
+    + [Amplify Design](#amplify-design)
+  * [Getting Started](#getting-started)
+  * [Tools](#tools)
+  * [Workflows](#workflows)
+    + [Adding code to support a new feature](#adding-code-to-support-a-new-feature)
+  * [Reporting Bugs/Feature Requests](#reporting-bugs-feature-requests)
+  * [Contributing via Pull Requests](#contributing-via-pull-requests)
 
-## Reporting Bugs/Feature Requests
 
-We welcome you to use the GitHub issue tracker to report bugs or suggest features.
+## Amplify History and Ethos
```

Comment:
This section is interesting reading. I wonder if there's another place for this sub-section, such as a `HISTORY.md` that we can link to from this document. Most folks who come to a contributing guide are often familiar with the project and looking for a bit more tactic nuts-and-bolts-y guidance. React has a "Design Principles" section at the end of their contributor guide which seems similar in intent which may be a pattern worth following.

Reply:
Yes ultimately this is probably what we should do, as well as a separate architecture page.

Reply:
Hm, okay. Yea, I'm 100% in this camp, now, too. I just audited a few other contributors guides for some open source projects. Not one had a section like this. I'm going to simply omit it from this PR. We can come back and find a place for this as part of later/separate work.